app/controllers/loans_controller.rb

```diff
@@ -0,0 +1,33 @@
+class LoansController <  ApplicationController
+  before_action :set_loan, only: [:show]
+  def new
+    @loan = Loan.new
+  end
+
+  def create
+    @loan = Loan.create(loan_params)
+    if @loan.save
+      redirect_to @loan
+    end
+  end
+
+  def show
+    @loan = Loan.new
+    @payments = Payment.final_payment_list(@seted_loan.loan_amount.to_f,
```

Comment:
This logic is way too heavy to be in the controller. I suggest the use of serializers here, they are a class where you can organize the response, and here on the controller you would be only calling the serializer class. I can help you with that if you like ✌️ 
Here you can also find the [docs](https://github.com/rails-api/active_model_serializers/tree/v0.10.6/docs)

Reply:
Sure, I will love to know more.